plasma/root_chain/contracts/RootChain/RootChain.sol

```diff
@@ -193,15 +194,32 @@ contract RootChain {
         exit memory currentExit = exits[exitsQueue.getMin()];
         uint256 blknum = currentExit.utxoPos.div(1000000000);
         while (childChain[blknum].created_at < twoWeekOldTimestamp && exitsQueue.currentSize() > 0) {
-            currentExit.owner.transfer(currentExit.amount);
+            // currentExit.owner.transfer(currentExit.amount);
+            // Prevents attacker from submitting an address of a malicous contract that could revert transfer()
+            balances[currentExit.owner] = balances[currentExit.owner].add(currentExit.amount);
             uint256 priority = exitsQueue.delMin();
             delete exits[priority];
             delete exitIds[currentExit.utxoPos];
             currentExit = exits[exitsQueue.getMin()];
         }
     }
 
-    /* 
+    // @dev Allows any to withdraw their funds from the contract (eg. exit refunds, bonds, etc.)
+    // @dev Isolates transfers so attacker can't halt essential processes by reverting transfer()
+    function withdraw()
+        public
+        returns (uint256)
+    {
+        if (balances[msg.sender] == 0) {
```

Comment:
Would you mind adding this check as a `require` to `startExit`?

Reply:
Can you clarify what you mean?
- Check that the balance is 0 before someone initiates a new `startExit()`?
- Or check that the amount in `txList` is > 0?